### DISCUSSION TITLE
部署完后报错

### DISCUSSION THREAD

**Discussion by liangfyx** [spencerwooo/onedrive-vercel-index#240] [Q&A] [answered]
小白问题，已经配置好Redis数据库，但是打开页面后报错，重新部署了好多次还是不行，nodejs换12版本的也失败了，求解

![image](https://user-images.githubusercontent.com/20663717/148564803-3543f33e-c1c2-42a1-871b-7e2d64d9565a.png)
![image](https://user-images.githubusercontent.com/20663717/148565098-6c3860a8-7b20-4b6b-b89a-85278473e96c.png)

**Comment by spencerwooo** [comment, marked as answer]
我看你好像跟 https://github.com/spencerwooo/onedrive-vercel-index/discussions/238 的报错是一样的，所以也按照我在那边给出的解决办法排查一下？

> - configured your site.json, where the baseDirectory should be a folder in your root OneDrive path
> - the redis db url (if hosted on upstash, and has SSL enabled), should be something like rediss://:xxx@us1-xxx.upstash.io:xxx
notice the rediss instead of redis and the ://:xx instead of ://xx

**Comment by liangfyx** [comment]
按照上面教程修复完毕后又出现了500错误，我使用的是E5开发版的office订阅，账号为子管理员账号

补充：onedrive根目录下已经包含baseDirectory目录

错误日志如下：

<details>

```
[GET] /api?path=/
11:16:09:00
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      _header: 'GET /v1.0/me/drive/root:%2FPublic?select=%40microsoft.graph.downloadUrl,name,size,id,lastModifiedDateTime,folder,file HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Authorization: Bearer eyJ0eXAiOiJK...\r\n' +
        'User-Agent: axios/0.21.4\r\n' +
        'Host: graph.microsoft.com\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'GET',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/v1.0/me/drive/root:%2FPublic?select=%40microsoft.graph.downloadUrl,name,size,id,lastModifiedDateTime,folder,file',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'graph.microsoft.com',
      protocol: 'https:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    data: { error: [Object] }
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]
}
RequestId: e8d6b0e1-05d7-4041-b5e9-fa0441e96c9a Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

![image](https://user-images.githubusercontent.com/20663717/148714610-338de3e1-f71d-43d7-bf94-57e3031f5997.png)

</details>

**Reply by spencerwooo** [reply]
最新版本部署一下，新增了 error handling 有更详细的报错信息，根据你目前的信息不太好判断问题在哪。

**Reply by liangfyx** [reply]
你好现在报错如下，Functions log [里未报错，网站链接：](https://onedrive-vercel-index-m7jy126yr-liangfyx.vercel.app/)
`{"error":{"error":{"code":"itemNotFound","message":"The resource could not be found.","innerError":{"date":"2022-01-10T09:05:48","request-id":"90b11672-fd26-4fbe-9a57-a4dd9261c11a","client-request-id":"90b11672-fd26-4fbe-9a57-a4dd9261c11a"}}}}
`

**Reply by spencerwooo** [reply]
根目录设置错了呗，你的 OneDrive 文件夹里面真的有 Public 文件夹吗？你的 `"baseDirectory": "/Public"` 设置的就是这个。

**Reply by liangfyx** [reply]
谢谢，我之前文件夹名字叫baseDirectory，尴尬了，已解决
